### PR TITLE
Add transcript struct for compatibility

### LINKED ISSUE #11: Implement transcript in device side
Implement transcript in device code to connect implementation with other proof system implementations.

### PR DESCRIPTION
**Pull request by jeong0982**
This PR resolves #11 
## What does this PR change
- Add `Transcript` to cuda implementation
    - Run with member functions of `Transcript` such as `write_field_element`
    - todo: add hash function to `squeeze_challenge`
- Add `CudaTranscript` trait for generic use of transcript
    - Implementation of trait can choose hash function for challenges
- Add `CudaKeccakTranscript` struct

### CODE REVIEW

File: sumcheck/src/transcript.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    fn new(start: CudaSlice<u8>, cursor: usize, end: usize) -> Self {
+        Self { start, cursor, end }
+    }
+}
+
+pub fn from_u8_to_f<F: PrimeField + SerdeObject>(v: &[u8]) -> F {
+    F::from_raw_bytes_unchecked(v) * F::ONE
+}
+
+pub struct CudaKeccakTranscript<F> {
```
Comment (DoHoonKim8) [inline sumcheck/src/transcript.rs:40]: So I think we should declare `CudaKeccakTranscript` in C++ side, and let `bindgen` to generate Rust struct automatically for us. `FieldBinding` works in this way, would you check this?

File: sumcheck/src/cpu/sumcheck.rs [hunk 2]
```diff
@@ -74,3 +76,55 @@ pub(crate) fn verify_sumcheck<F: PrimeField>(
     }
     true
 }
+
+pub(crate) fn verify_sumcheck_transcript<F: PrimeField + halo2curves::serde::SerdeObject>(
```
[the remark this answers is not in the document]
Comment (jeong0982) [reply sumcheck/src/cpu/sumcheck.rs:80]: Okay, I'll remove it

File: sumcheck/src/gpu/cuda/kernels/multilinear.cu [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #include "../includes/barretenberg/ecc/curves/bn254/fr.cuh"
+#include <stdint.h>
```
[the remark this answers is not in the document]
Comment (jeong0982) [reply sumcheck/src/gpu/cuda/kernels/multilinear.cu:2]: Removed

**Comment by jeong0982** [comment]
Due to issue about memory, close this PR

**Pull request closed without merging**